src/ContentValidationListener.php

```diff
@@ -213,7 +215,35 @@ public function onRoute(MvcEvent $e)
         }
 
         $inputFilter->setData($data);
+
         if ($inputFilter->isValid()) {
+            if (isset($this->config[$controllerService]['use_raw_data'])
+                    && $this->config[$controllerService]['use_raw_data'] === true) {
+                $dataContainer->setBodyParams($data);
+
+                return;
+            }
+
+            if (!$inputFilter instanceof BaseInputFilter) {
+                return;
```

Comment:
By the time we get here, we know we want to use the data from the input filter; shouldn't we set the values we have in the `$dataContainer`? The only reason for the `BaseInputFilter` test is so we know whether or not to merge the unknown values as well.